### PR TITLE
DM-13950: Convert assembleCoadd.py to numpydoc

### PR DESCRIPTION
**Pull request by mrawls**
The scope of this ticket changed to update only assembleCoadd.py with numpydoc for now.

### CODE REVIEW

File: doc/lsst.pipe.tasks/index.rst [hunk 1]
```diff
@@ -12,3 +12,6 @@ Python API reference
 ====================
 
 .. automodapi:: lsst.pipe.tasks
+   :skip: AssembleCoaddTask, SafeClipAssembleCoaddTask, CompareWarpAssembleCoaddTask
```
Comment (jonathansick) [inline doc/lsst.pipe.tasks/index.rst]: To get the API docs to show up, you'll need to change this to:

```
.. automodapi:: lsst.pipe.tasks.assembleCoadd
```

We'll generalize it to all of `lsst.pipe.tasks` when those APIs are available.

File: python/lsst/pipe/tasks/assembleCoadd.py [hunk 2]
```diff
@@ -938,169 +1032,168 @@ def validate(self):
...
+
+       assembleCoadd.py --help
+
+    To demonstrate usage of the `SafeClipAssembleCoaddTask` in the larger
+    context of multi-band processing, we will generate the HSC-I & -R band
+    coadds from HSC engineering test data provided in the ci_hsc package.
+    To begin, assuming that the lsst stack has been already set up, we must
+    set up the obs_subaru and ci_hsc packages. This defines the environment
+    variable $CI_HSC_DIR and points at the location of the package. The raw
+    HSC data live in the ``$CI_HSC_DIR/raw``h v directory. To begin assembling
```
Comment (jonathansick) [inline python/lsst/pipe/tasks/assembleCoadd.py]: This was generating a build error. Should be:

```
HSC data live in the ``$CI_HSC_DIR/raw`` directory. To begin assembling
```

File: doc/lsst.pipe.tasks/index.rst (hunk 1 above)
Comment (mrawls) [reply doc/lsst.pipe.tasks/index.rst]: Two questions here. (1) I was originally going to have just assembleCoadd listed, but it looks like @r-owen recently added all of lsst.pipe.tasks instead. Perhaps there's some special case I'm missing? (2) Does your comment here affect the `:skip:` line or just the `.. automodapi::` line?

File: COPYRIGHT [hunk 3]
```diff
@@ -0,0 +1 @@
+Copyright (c) 2008-2018 Association of Universities for Research in Astronomy, Inc.
```
[the remark this answers is not in the document]
Comment (mrawls) [reply COPYRIGHT]: Indeed. Is this an acceptable interim option, or should I add a note saying the copyright file is incomplete or something? Happy to make a ticket for it.

File: doc/lsst.pipe.tasks/index.rst (hunk 1 above)
Comment (jonathansick) [reply doc/lsst.pipe.tasks/index.rst]: The comment is a replacement for both lines.

`lsst.pipe.tasks` doesn't import anything, turns out, so we need to point directly at individual modules

File: COPYRIGHT (hunk 3 above)
[the remark this answers is not in the document]
Comment (jonathansick) [reply COPYRIGHT]: I... wouldn't make a ticket just for this. I think it's understood that copyrights need to be systematically audited everywhere given the new system so there's gotta be an overarching ticket for all this anyways. I'll figure out the who and how of this work.

**Pull request merged**